nilearn/connectome/connectivity_matrices.py

```diff
@@ -535,7 +539,13 @@ def _fit_transform(
 
         # casting to a list
         # to make it easier to check with sklearn estimator compliance
-        if not isinstance(X, list):
+        if not isinstance(X, (np.ndarray, list)):
+            raise TypeError(
+                "Input must be list or numpy array. "
+                f"Got {X.__class__.__name__}"
+            )
+        if isinstance(X, np.ndarray) and X.ndim == 2:
```

Comment:
In practice this means that 3D arrays are accepted as input but it's not mention in the doc string.

Reply:
ok changed the code and the doc strings to accept iterable of numpy arrays

Reply:
not quite because check_input does not modify X